### PR TITLE
Assessment status change: Show recipient list, send emails for current cycle only

### LINKED ISSUE #2035: BUG- status change email(only read by people from the current cycle)
@minotogna 
Status change emails can **only** be read by people(NCs, collaborators, etc...) of the ****current** cycle**, **not including** the previous cycle(s).

e.g. People from last reporting cycle(2020) can also receive the status change email for this cycle(2025).

### PR DESCRIPTION
**Pull request by sorja**
Resolves #2035 

- Add copy link to invitation
- Display recipients
- Only send for current cycle

Status change recipients:

https://user-images.githubusercontent.com/5508251/213930507-14c95d02-b2c7-4c64-a43b-bbc8fcd79bca.mov

Copy:

https://user-images.githubusercontent.com/5508251/213926845-8f2bfd5d-70c4-455d-91fe-af1513c83e41.mov

## Styling update: bring back to what it was; video showing old styling above

image

### CODE REVIEW

File: src/server/repository/public/user/readCountryUsersByRole.ts [hunk 1]
```diff
@@ -15,20 +16,20 @@ import { fields } from './fields'
...
 ): Promise<Array<User>> => {
-  const { countryISOs, roles } = props
+  const { countryISOs, cycle, roles } = props
 
   return client.map<User>(
     `
         select ${selectFields}, jsonb_agg(to_jsonb(ur.*)) as roles
         from public.users u join users_role ur on (u.id = ur.user_id)
-        where ur.country_iso in ($1:csv) and ur.role in ($2:csv)
+        where ur.country_iso in ($1:csv) and ur.role in ($2:csv) and ur.cycle_uuid = $3
```
Comment (sorja) [inline src/server/repository/public/user/readCountryUsersByRole.ts:28]: Actual fix (user role must exist for current cycle)

**Comment by minotogna** [comment]
I noticed the modal has changed layout . Can you please fix the size of the modal to how it was?

FRA platform

File: src/client/components/UserList/UserInvitationInfo/UserInvitationInfo.scss [hunk 2]
```diff
@@ -13,6 +13,11 @@
   justify-content: space-between;
   padding: 14px;
```
Comment (minotogna) [inline src/client/components/UserList/UserInvitationInfo/UserInvitationInfo.scss]: Dont we have a generic btn-transparent class ?

File: src/meta/user/userRoles.ts [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
 
 const noRole = { role: 'NONE', labelKey: 'user.roles.noRole' }
 
+// Return roles to receive email on country assessment status change
+const getRecipientRoles = (props: { status: AssessmentStatus }) => {
+  const { status } = props
+
+  switch (status) {
+    case AssessmentStatus.editing:
+      return [RoleName.NATIONAL_CORRESPONDENT]
```
Comment (minotogna) [inline src/meta/user/userRoles.ts:16]: Dont NC, C, ANC here and at line 22 should receive an email as well?

File: src/client/components/PageLayout/Toolbar/Status/useRecipients.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import { useCountryIso, useGetRequest } from '@client/hooks'
+
+import { StatusTransition } from './types'
+
+export const useRecipients = (props: { status: StatusTransition }): Array<User> => {
+  const { status } = props
+  const countryIso = useCountryIso()
+  const cycle = useCycle()
+  const assessment = useAssessment()
+  const { data, dispatch: fetchData } = useGetRequest(ApiEndPoint.User.countryStatusChangeRecipients(), {
```
Comment (minotogna) [inline src/client/components/PageLayout/Toolbar/Status/useRecipients.ts]: Any chance we can use users like in country home without adding this new endpoint ?

File: src/client/components/UserList/UserInvitationInfo/UserInvitationInfo.scss (hunk 2 above)
Comment (sorja) [reply src/client/components/UserList/UserInvitationInfo/UserInvitationInfo.scss]: no, added

File: src/meta/user/userRoles.ts (hunk 3 above)
Comment (sorja) [reply src/meta/user/userRoles.ts:16]: to be discussed

File: src/client/components/PageLayout/Toolbar/Status/useRecipients.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { User } from '@meta/user'
+import { UserRoles } from '@meta/user/userRoles'
+
+import { useUsers } from '@client/store/ui/userManagement'
+import useGetUsers from '@client/pages/AssessmentHome/hooks/useGetUsers'
+
+import { StatusTransition } from './types'
+
+export const useRecipients = (props: { status: StatusTransition }): Array<User> => {
+  useGetUsers()
```
Comment (minotogna) [inline src/client/components/PageLayout/Toolbar/Status/useRecipients.ts]: maybe we should just invoke this hook only once in Country page? (and not in home + here) what do you think?

File: src/server/api/user/index.ts [hunk 6]
```diff
@@ -8,16 +8,16 @@ import { acceptInvitation } from './acceptInvitation'
...
 import { invite } from './invite'
 import { removeInvitation } from './removeInvitation'
 import { sendInvitationEmail } from './sendInvitationEmail'
 import { updateSectionAuth } from './updateSectionAuth'
 import { updateUser } from './updateUser'
-import multer = require('multer')
-import { getResetPassword } from './getResetPassword'
 import { updateUserAdminRole } from './updateUserAdminRole'
 import { updateUserRoles } from './updateUserRoles'
+import multer = require('multer')
```
Comment (minotogna) [inline src/server/api/user/index.ts]: can't we import from ? instead of require ?

**Review by rfontanarosa — DISMISSED** [review]

File: src/client/pages/Country/Country.tsx [hunk 7]
```diff
@@ -18,6 +18,7 @@ import { useCountryIso, useIsDataExportView } from '@client/hooks'
 import Navigation from '@client/components/Navigation'
 import AssessmentDataDownload from '@client/pages/AssessmentDataDownload'
 import AssessmentHome from '@client/pages/AssessmentHome'
+import useGetUsers from '@client/pages/AssessmentHome/hooks/useGetUsers'
```
Comment (minotogna) [inline src/client/pages/Country/Country.tsx]: I would also move thebhook here

Comment (sorja) [reply src/client/pages/Country/Country.tsx]: Good catch! Moved

**Review by minotogna — APPROVED** [review]

**Review by rfontanarosa — APPROVED** [review]

**Pull request merged**